src/data_factory_testing_framework/functions/evaluator/rules/expression_rule.py

```diff
@@ -0,0 +1,44 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Union
+
+from lark import Tree
+
+from data_factory_testing_framework.functions.evaluator.exceptions import ExpressionEvaluationInvalidChildTypeError
+
+
+@dataclass
+class EvaluatedExpression:
```

Comment:
The name `EvaluatedExpression` feels a bit counter-intuitive in a number of places. It can represent an evaluated expression, but it can also represent a plain literal. Maybe we can brainstorm about this one together.

Reply:
Yes - wasn't really pleased with the naming either. Indeed, it represents literal values and evaluated values. In the end it's actual values that are derived from evaluating the parsed lark tree, thus my naming. Happy to hear suggestions.

Reply:
Suggestion of brainstorm: EvaluationResult